tests/integration/enc/models/test_alexnet.py

```diff
@@ -37,21 +35,20 @@ def test_main(self, enc_asset_loader):
         desired = asset.output.enc_keys
         assert actual == desired
 
-    def test_state_dict_url(self, subtests, frameworks):
+    def test_state_dict_url(self, frameworks):
         def should_be_available(framework):
             return framework == "torch"
 
         multi_layer_encoder = enc.alexnet_multi_layer_encoder(pretrained=False)
 
         for framework in frameworks:
-            with subtests.test(framework=framework):
-                if should_be_available(framework):
-                    assert isinstance(
-                        multi_layer_encoder.state_dict_url(framework), str
-                    )
-                else:
-                    with pytest.raises(RuntimeError):
-                        multi_layer_encoder.state_dict_url(framework)
+            if should_be_available(framework):
+                assert isinstance(
+                    multi_layer_encoder.state_dict_url(framework), str
+                )
+            else:
+                with pytest.raises(RuntimeError):
+                    multi_layer_encoder.state_dict_url(framework)
```

Comment:
By just removing `subtests` here, we lose information. If this test now fails, we can't know which `framework` caused this without manually debugging it. Every time you see `subtests` used in conjunction with a loop, you can probably replace it with a test parametrization. In this case for example:

```python
    @pytest.mark.parametrize(
        ("framework", "should_be_available"),
        [
            pytest.param(framework, should_be_available, id=framework)
            for framework, should_be_available in [("torch", True), ("caffe", False)]
        ],
    )
    def test_state_dict_url(self, framework, should_be_available):
        multi_layer_encoder = enc.alexnet_multi_layer_encoder(pretrained=False)

        if should_be_available:
            assert isinstance(multi_layer_encoder.state_dict_url(framework), str)
        else:
            with pytest.raises(RuntimeError):
                multi_layer_encoder.state_dict_url(framework)
```

What this does, is that the test will actually be instantiated two times. This way we can easily see which parametrization is failing and have successfully removed the `subtests`.